x-pack/plugins/actions/server/application/connector/methods/list_types/list_types.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ActionsClientContext } from '../../../../actions_client';
+import { ConnectorType } from '../../types';
+import { ListTypesParams } from './types';
+
+export async function listTypes(
+  context: ActionsClientContext,
+  { featureId, includeSystemActionTypes }: ListTypesParams
+): Promise<ConnectorType[]> {
+  const connectorTypes = context.actionTypeRegistry.list(featureId);
```

Comment:
nitpick: Not sure if we wanted to validate the params schema here, we do it in the rules client methods because it is reused by other solutions (even though the same params are validated at the route level).

Reply:
Something like this? https://github.com/elastic/kibana/pull/162830/commits/147a171461556e4097a8e85574b8299a78f767ca

Reply:
Sort of, I'm thinking something more akin to a run time type check using schema.validate

Reply:
Gotcha...so like this? https://github.com/elastic/kibana/pull/162830/commits/8c254014a9dff235ffed39d1f6d89ac413e773c9 🤞